lxd/seccomp/seccomp.go

```diff
@@ -1675,6 +1693,138 @@ func (s *Server) HandleSchedSetschedulerSyscall(c Instance, siov *Iovec) int {
 	return 0
 }
 
+// HandleSysinfoSyscall handles sysinfo syscalls.
+func (s *Server) HandleSysinfoSyscall(c Instance, siov *Iovec) int {
+	l := logger.AddContext(logger.Log, logger.Ctx{
+		"container":             c.Name(),
+		"project":               c.Project(),
+		"syscall_number":        siov.req.data.nr,
+		"audit_architecture":    siov.req.data.arch,
+		"seccomp_notify_id":     siov.req.id,
+		"seccomp_notify_flags":  siov.req.flags,
+		"seccomp_notify_pid":    siov.req.pid,
+		"seccomp_notify_fd":     siov.notifyFd,
+		"seccomp_notify_mem_fd": siov.memFd,
+	})
+
+	defer l.Debug("Handling sysinfo syscall")
+
+	info := unix.Sysinfo_t{}
+	err := unix.Sysinfo(&info)
+	if err != nil {
+		l.Warn("Failed getting sysinfo", logger.Ctx{"err": err})
+		C.seccomp_notify_update_response(siov.resp, 0, C.uint32_t(seccompUserNotifFlagContinue))
+
+		return 0
+	}
+
+	cg, err := cgroup.NewFileReadWriter(int(siov.msg.init_pid), liblxc.HasApiExtension("cgroup2"))
+	if err != nil {
+		l.Warn("Failed loading cgroup", logger.Ctx{"err": err, "pid": siov.msg.init_pid})
+		C.seccomp_notify_update_response(siov.resp, 0, C.uint32_t(seccompUserNotifFlagContinue))
+
+		return 0
+	}
+
+	// Get instance uptime.
+	f, err := os.Stat(fmt.Sprintf("/proc/%d", siov.msg.init_pid))
```

Comment:
@brauner @stgraber is this an acceptable way to get the instance's uptime? It works, but not sure if there is a better way.

Reply:
Yep, that's how we do it in LXCFS